### PR TITLE
feat(examples): expose ownable & pausable safe objects, update ownable API

### PR DESCRIPTION
**Pull request by leohhhn**
## Description

This PR exposes safe objects where possible for ownable and pausable packages.

Let's start exposing variables named `Ownable`, ie:

```go
var Ownable = ownable.New()
```

This is the intended use, as exposing this field allows direct, safe, MsgRun calls to access the ownable functions. It's also impossible to directly reassign the value of this variable from another realm.

This PR also introduces a BREAKING CHANGE: the `Ownable.CallerIsOwner` API now returns a boolean instead of an error, which makes more sense considering the name of the function.

### CODE REVIEW

**Comment by Gno2D2** [comment]
#### 🛠 PR Checks Summary
All **Automated Checks** passed. ✅

##### Manual Checks (for Reviewers):
- [ ] **SKIP**: Do not block the CI for this PR


<details><summary>Read More</summary>

🤖 This bot helps streamline PR reviews by verifying automated checks and providing guidance for contributors and reviewers.

##### ✅ Automated Checks (for Contributors):
 🟢 Maintainers must be able to edit this pull request ([more info](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork))
 🟢 The pull request head branch must be up-to-date with its base ([more info](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/keeping-your-pull-request-in-sync-with-the-base-branch))


##### ☑️ Contributor Actions:
1. Fix any issues flagged by automated checks.
2. Follow the Contributor Checklist to ensure your PR is ready for review.
	- Add new tests, or document why they are unnecessary.
	- Provide clear examples/screenshots, if necessary.
	- Update documentation, if required.
	- Ensure no breaking changes, or include `BREAKING CHANGE` notes.
	- Link related issues/PRs, where applicable.

##### ☑️ Reviewer Actions:
1. Complete manual checks for the PR, including the guidelines and additional checks if applicable.

##### 📚 Resources:
- [Report a bug with the bot](https://github.com/gnolang/gno/issues/3238).
- [View the bot’s configuration file](https://github.com/gnolang/gno/tree/master/contribs/github-bot/internal/config/config.go).

<details><summary><b>Debug</b></summary><blockquote>
<details><summary><b>Automated Checks</b></summary><blockquote>

<details><summary>Maintainers must be able to edit this pull request (more info)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 The pull request was created from a fork (head branch repo: leohhhn/gno)

```
### Then
```
🟢 Requirement satisfied
└── 🟢 Maintainer can modify this pull request

```
</blockquote></details>

<details><summary>The pull request head branch must be up-to-date with its base (more info)</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 On every pull request

```
### Then
```
🟢 Requirement satisfied
└── 🟢 Head branch (leohhhn:ownable-pausable) is up to date with base (master): behind by 0 / ahead by 2

```
</blockquote></details>

</blockquote></details>


<details><summary><b>Manual Checks</b></summary><blockquote>

<details><summary>**SKIP**: Do not block the CI for this PR</summary><blockquote>

### If
```
🟢 Condition met
└── 🟢 On every pull request

```
### Can be checked by

- Any user with comment edit permission

</blockquote></details>

</blockquote></details>

</blockquote></details>

</details>

File: examples/gno.land/p/demo/memeland/memeland.gno [hunk 1]
```diff
@@ -160,8 +160,8 @@ func (m *Memeland) RemovePost(id string) string {
 		panic("id cannot be empty")
 	}
 
-	if err := m.CallerIsOwner(); err != nil {
-		panic(err)
+	if !m.CallerIsOwner() {
+		panic(ownable.ErrUnauthorized.Error())
```
Comment (gfanton) [inline examples/gno.land/p/demo/memeland/memeland.gno:164]: nit: does this works ?
```suggestion
		panic(ownable.ErrUnauthorized)
```

File: examples/gno.land/r/gnoland/monit/monit.gno [hunk 2]
```diff
@@ -20,7 +20,7 @@ var (
 	lastUpdate       time.Time
 	lastCaller       std.Address
 	wd               = watchdog.Watchdog{Duration: 5 * time.Minute}
-	owner            = ownable.New() // TODO: replace with -> ownable.NewWithAddress...
+	OwnerStore       = ownable.New() // TODO: replace with -> ownable.NewWithAddress...
```
Comment (gfanton) [inline examples/gno.land/r/gnoland/monit/monit.gno:23]: I think there is a naming problem here. Why change this? `Owner` seems correct to me; it doesn't look like a store. Additionally, it's weird to end with `OwnerStore` when calling `ownable.New`. I suggest either renaming it to `ownable.NewStore` or changing the variable itself. However, based on what `New` actually returns, it doesn't make too much sense to refer to it as a store.

Comment (leohhhn) [reply examples/gno.land/r/gnoland/monit/monit.gno:23]: Store was the best I can come up with; ideally you could just call it `ownable` but that conflicts with the pkgname. `Owner` conflicts with the ownable.owner field. it's kind of weird; and yes, `store` might not be the best naming.

Comment (leohhhn) [reply examples/gno.land/r/gnoland/monit/monit.gno:23]: But Admin & Pause also don't ring the correct bells, not sure what to do.

Comment (leohhhn) [reply examples/gno.land/r/gnoland/monit/monit.gno]: Switched to:

```go
var Ownable = ownable.New()
```

This seems optimal

**Review by gfanton — APPROVED** [review]
LGTM 👍

**Review by n2p5 — APPROVED** [review]
Clean and easy to follow PR, much appreciated on that. LGTM

**Pull request merged**